hack/update-codegen.sh

```diff
@@ -18,21 +18,34 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-cd "$(dirname "${0}")"
+cd "$(dirname "${0}")/.."
 GO_CMD=${1:-go}
-CODEGEN_PKG=${2:-../bin}
+CODEGEN_PKG=${2:-bin}
 REPO_ROOT="$(git rev-parse --show-toplevel)"
 
 echo "GOPATH=$GOPATH"
 
-chmod +x "${CODEGEN_PKG}/generate-internal-groups.sh"
+source "${CODEGEN_PKG}/kube_codegen.sh"
 
-bash "${CODEGEN_PKG}/generate-groups.sh" \
-  "applyconfiguration,client,informer,lister" \
-  sigs.k8s.io/jobset/client-go \
-  sigs.k8s.io/jobset/api \
-  jobset:v1alpha2 \
-  --go-header-file ./boilerplate.go.txt
+# TODO: remove the workaround when the issue is solved in the code-generator
+# (https://github.com/kubernetes/code-generator/issues/165).
+# Here, we create the soft link named "sigs.k8s.io" to the parent directory of
+# Kueue to ensure the layout required by the kube_codegen.sh script.
+ln -s .. sigs.k8s.io
+trap "rm sigs.k8s.io" EXIT
+
+kube::codegen::gen_helpers \
+    --input-pkg-root sigs.k8s.io/jobset/api \
+    --output-base "${REPO_ROOT}" \
+    --boilerplate "${REPO_ROOT}/hack/boilerplate.go.txt"
+
+kube::codegen::gen_client \
+    --with-watch \
+    --with-applyconfig \
+    --input-pkg-root sigs.k8s.io/jobset/api \
+    --output-base "$REPO_ROOT" \
+    --output-pkg-root sigs.k8s.io/jobset/client-go \
+    --boilerplate "${REPO_ROOT}/hack/boilerplate.go.txt"
 
 # if client-go files were generated outside of repo root, attempt to move them to the repo root.
 CLIENT_GO=$(find $GOPATH -regextype sed -regex ".*jobset.*client-go")
```

Comment:
I'm not sure if we need this part anymore. Can anyone please confirm?

Reply:
Why do you think it is not needed? In the past we've encountered issues where the generated client-go files appear in different locations in the filesystem in different environments, so we have this logic to move the files to the correct location if they end up somewhere in $GOROOT

Reply:
@ashnehete change is trying to fix that issue.  

Both Kueue and Jobset ran into this issue: https://github.com/kubernetes/code-generator/issues/165.  

So I guess the solution is to add a symbolic link for CWD into sigs.k8s.io.  

So what this change does is we should be generating everything into our CWD, but I guess we trick codegen to assume that its actually in sigs.k8s.io.

Reply:
I see, in that case yes feel free to remove this logic once you've tested manually and confirmed the symbolic link trick works.

Reply:
The new kube_codegen.sh script seems to be enforcing the output location as specified. I have tested the code manually on MacOS and Ubuntu (Docker).

Reply:
I also pulled this branch and did a quick test with my setup.  I added a dummy api and I was able to confirm that this generates both client-go and python code.  

I don't think I tested any changes where the client-go copying was disabled though.